certification/internal/shell/has_minimal_vulns.go

```diff
@@ -1,16 +1,96 @@
 package shell
 
 import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
-	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/utils/file"
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	ovalFilename = "rhel-8.oval.xml.bz2"
+	ovalUrl      = "https://www.redhat.com/security/data/oval/v2/RHEL8/"
+	reportFile   = "vuln.html"
+)
+
 type HasMinimalVulnerabilitiesCheck struct{}
 
 func (p *HasMinimalVulnerabilitiesCheck) Validate(image string, logger *logrus.Logger) (bool, error) {
-	return false, errors.ErrFeatureNotImplemented
+
+	ovalFileUrl := fmt.Sprintf("%s%s", ovalUrl, ovalFilename)
+
+	tempdir, err := os.MkdirTemp(os.TempDir(), "oval-*")
+	if err != nil {
+		logger.Error("Unable to create temp dir", err)
+		return false, err
+	}
+	logger.Debugf("Oval file dir: %s", tempdir)
+	defer os.RemoveAll(filepath.Dir(tempdir))
```

Comment:
Two things here:

1) I don't think you need `file path.Dir(...)` here. This is deleting my `/tmp` which is causing the test to fail in subsequent executions. See [1] below for more details on that.

2) `os.RemoveAll` can throw an error, which makes it not a great candidate for defer. We probably want to let the user know that we couldn't Clea up in this case.

---
Thinking on this - no action needed for this PR as a result of this, just food for thought/discussion:
I wonder if we could figure out a way to elevate our permissions _specifically_ when calling out to oscap-podman. Just seems like a safer approach than calling preflight itself with sudo. As a small scale example, removing `/tmp` probably wouldn't have succeeded if I hadn't called `sudo preflight ...`. 

Not all checks require elevated permissions. So either this check is executed on its own to ensure that all _other_ checks aren't running with more privileges than they need, or we run them all with elevated privileges (what I'm thinking we should avoid doing).

---
[1]

I updated `Validate` to just print what file we were targeting like so:

```go
	logger.Debugf("Oval file dir: %s", tempdir)
	logger.Debug("debug!!!!", filepath.Dir(tempdir)) // printline debugging here, woohoo!
	defer os.RemoveAll(filepath.Dir(tempdir))
```

I got this in the log:

```
time="2021-07-07T10:18:12-05:00" level=debug msg="debug!!!!/tmp"
```

Here's the error that tipped me off that we were removing more than we should have

```
time="2021-07-07T10:18:04-05:00" level=info msg="check completed: HasMinimalVulnerabilities" error="stat /tmp: no such file or directory" result=ERROR
```

Reply:
Thank you Jose for the through investigation... Appreciate your time and effort :) Will fix the issue of removing `/tmp`.

I'll take a look into the `sudo` thing and keep you posted if I found anything.